fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt

```diff
@@ -161,3 +194,31 @@ private fun generateFakeTabsList(tabCount: Int = 10, isPrivate: Boolean = false)
             ),
         )
     }
+
+@SuppressWarnings("EmptyFunctionBlock")
+private fun createFakeInteractor(): TabsTrayInteractor =
+    object : TabsTrayInteractor {
+        override fun onTabSelected(tab: TabSessionState, source: String?) {}
+        override fun onTabClosed(tab: TabSessionState, source: String?) {}
+        override fun onSyncedTabClicked(tab: Tab) {}
+        override fun onTrayPositionSelected(position: Int, smoothScroll: Boolean) {}
+        override fun onDeletePrivateTabWarningAccepted(tabId: String, source: String?) {}
+        override fun onDeleteTabs(tabs: Collection<TabSessionState>) {}
+        override fun onInactiveDebugClicked(tabs: Collection<TabSessionState>) {}
+        override fun onTabsMove(tabId: String, targetId: String?, placeAfter: Boolean) {}
+        override fun onRecentlyClosedClicked() {}
+        override fun onMediaClicked(tab: TabSessionState) {}
+        override fun onMultiSelectClicked(tab: TabSessionState, source: String?) {}
+        override fun onTabLongClicked(tab: TabSessionState): Boolean = true
+        override fun onBackPressed(): Boolean = true
+        override fun onTabUnselected(tab: TabSessionState) {}
+        override fun onInactiveTabsHeaderClicked(expanded: Boolean) {}
+        override fun onInactiveTabClicked(tab: TabSessionState) {}
+        override fun onInactiveTabClosed(tab: TabSessionState) {}
+        override fun onDeleteAllInactiveTabsClicked() {}
+        override fun onAutoCloseDialogCloseButtonClicked() {}
+        override fun onEnableAutoCloseClicked() {}
+        override fun onNormalTabsFabClicked() {}
+        override fun onPrivateTabsFabClicked() {}
+        override fun onSyncedTabsFabClicked() {}
+    }
```

Comment:
This could be avoided by redefining the composable to take lambda parameters instead of the entire interactor. It results in annoyingly long parameter lists, but it also decouples the view better from the class handling its logic. It also follows established precedent elsewhere in the app